### PR TITLE
Deployment related actions

### PR DESCRIPTION
**Pull request by lemaitre-aneo**
Defines the following actions:
- install dependencies, eg: terraform, k3s, aws cli
- deploy: either localhost or aws
- destroy: either localhost or aws
- bootstrap deploy: deploy a tfstate bucket on aws
- bootstrap destroy: destroy a tfstate bucket on aws

The deployment can be configured to use different armonik image versions if necessary.

### CODE REVIEW

**Comment by lemaitre-aneo** [comment]
The goal is to have a first `v1` for the actions, so I would really like to have these actions as close to stable as possible.

File: bootstrap-deploy/action.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: "Destroy ArmoniK deployment"
```
Comment (esoubiran-aneo) [inline bootstrap-deploy/action.yml]: Le nom n'est pas le bon (pas en accord avec le dossier dans tous les cas). Il doit s'agit plus d'un `deploy`

File: dependencies/action.yml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        sudo systemctl enable containerd.service
+
+        docker run hello-world
+        
+    - if: ${{ inputs.k3s && inputs.k3s != 'false' }}
+      name: Install K3s
+      shell: bash
+      run: |
+        set -ex
+        curl -sfL https://get.k3s.io | INSTALL_K3S_VERSION="v1.23.8+k3s1" sh -s - --write-kubeconfig-mode 644 --docker --write-kubeconfig ~/.kube/config
```
Comment (aneojgurhem) [inline dependencies/action.yml]: parameter for Kubernetes version ?

File: dependencies/action.yml [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+    - name: Install Kubectl
+      shell: bash
+      run: |
+        set -ex
+        curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+        curl -LO "https://dl.k8s.io/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl.sha256"
+        echo "$(cat kubectl.sha256) kubectl" | sha256sum --check
+        mkdir -p ~/.kube
+        sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+    - if: ${{ inputs.aws && inputs.aws != 'false' }}
```
Comment (aneojgurhem) [inline dependencies/action.yml]: Those conditions seem redundant

Comment (lemaitre-aneo) [reply dependencies/action.yml]: They seems, but are not. The problem is: Github Action supports boolean types, but not in custom actions. The way the test is written ensures it works now on string parameters, and in the future if they decide to allow typed parameters.

File: dependencies/action.yml (hunk 2 above)
Comment (lemaitre-aneo) [reply dependencies/action.yml]: Indeed, will fix

File: dependencies/action.yml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+    description: "Install K3s"
+    type: boolean
+    required: false
+    default: false
+  terraform:
+    description: "Install Terraform"
+    type: boolean
+    required: false
+    default: false
+  AWS_ACCESS_KEY_ID:
```
Comment (aneojgurhem) [inline dependencies/action.yml]: I think there is also a type or option for secrets.

Comment (lemaitre-aneo) [reply dependencies/action.yml]: There is for nested workflows, but not for custom actions.

Comment (aneojgurhem) [reply dependencies/action.yml]: desapointing...

File: dependencies/action.yml (hunk 3 above)
Comment (aneojgurhem) [reply dependencies/action.yml]: ok !

File: bootstrap-deploy/action.yml [hunk 5]
```diff
@@ -0,0 +1,27 @@
+name: "Bootstrap deploy"
+description: "Action to dieploy S3 for deployment tfstates"
```
Comment (aneojgurhem) [inline bootstrap-deploy/action.yml]:
```suggestion
description: "Action to deploy S3 for deployment tfstates"

```

File: destroy/action.yml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        [ "$EKS_NAME" != null ] || EKS_NAME=
+        [ "$EKS_REGION" != null ] || EKS_REGION=
+        if [ -n "$EKS_NAME" -a -n "$EKS_REGION" ]; then
+          aws eks update-kubeconfig --region "$EKS_REGION" --name "$EKS_NAME"
+          cat ~/.kube/config
+        else
+          # Ignore the update_kubeconfig if there is no more EKS
+          TF_DATA_DIR=generated terraform state rm module.eks.null_resource.update_kubeconfig || true
+        fi
+        make PREFIX="$INPUT_PREFIX" delete
```
Comment (aneojgurhem) [inline destroy/action.yml]: some files do not have empty lines at the end^^

**Review by aneojgurhem — DISMISSED** [review]

**Review by aneojgurhem — APPROVED** [review]

**Pull request merged**